pkg/config/polling_manager.go

```diff
@@ -89,21 +97,40 @@ func InitialDatafile(datafile []byte) OptionFunc {
 func (cm *PollingProjectConfigManager) SyncConfig(datafile []byte) {
 	var e error
 	var code int
+	var respHeaders http.Header
 
 	closeMutex := func(e error) {
 		cm.err = e
 		cm.configLock.Unlock()
 	}
 
 	if len(datafile) == 0 {
-		datafile, code, e = cm.requester.Get()
+		if cm.lastModified != "" {
+			lastModifiedHeader := utils.Header{Name: ModifiedSince, Value: cm.lastModified}
+			datafile, respHeaders, code, e = cm.requester.Get(lastModifiedHeader)
+		} else {
+			datafile, respHeaders, code, e = cm.requester.Get()
+		}
 
 		if e != nil {
 			cmLogger.Error(fmt.Sprintf("request returned with http code=%d", code), e)
 			cm.configLock.Lock()
 			closeMutex(e)
 			return
 		}
+
+		if code == 304 {
```

Comment:
you can use http.StatusNotModified instead of 304 ( it will be clearer)

Reply:
Updated all previously hardcoded status codes too.

Reply:
looks good, tested the code with datafile changes.